duchess-build-rs/src/shim_writer.rs

```diff
@@ -0,0 +1,102 @@
+use std::{fmt::Display, io::Write};
+
+use duchess_reflect::reflect::JavapClassInfo;
+
+use crate::code_writer::CodeWriter;
+
+pub struct ShimWriter<'w> {
+    cw: CodeWriter<'w>,
+    shim_name: &'w str,
+    java_interface_info: &'w JavapClassInfo,
+}
+
+impl<'w> ShimWriter<'w> {
+    pub fn new(
+        writer: &'w mut impl Write,
+        shim_name: &'w str,
+        java_interface_info: &'w JavapClassInfo,
+    ) -> Self {
+        ShimWriter {
```

Comment:
Do you think we should merge this as is or stash it in a branch until we are actually ready to use it?

Reply:
I am 99.9% sure I am going to want it :) I say just merge it.